driver/registry_memory.go

```diff
@@ -138,9 +145,30 @@ func (r *RegistryMemory) CredentialHandler() *api.CredentialsHandler {
 	return r.ch
 }
 
+func (r *RegistryMemory) HealthEventManager() health.EventManager {
+	if r.healthEventManager == nil {
+		r.healthEventManager = health.NewDefaultHealthEventManager()
+	}
+	return r.healthEventManager
+}
+
 func (r *RegistryMemory) HealthHandler() *healthx.Handler {
+	r.RLock()
+	defer r.RUnlock()
+
 	if r.healthxHandler == nil {
-		r.healthxHandler = healthx.NewHandler(r.Writer(), r.BuildVersion(), healthx.ReadyCheckers{})
+		rulesReadinessChecker := rulereadiness.NewReadinessHealthChecker()
+
+		healthCheckers := []health.Readiness{rulesReadinessChecker}
+
+		handlers := healthx.ReadyCheckers{}
+		for _, checker := range healthCheckers {
+			handlers[checker.Name()] = checker.Validate
+			if err := r.HealthEventManager().AddListener(checker); err != nil {
+				r.Logger().WithError(err).Fatalf("Unable to register health checker listener.")
+			}
+		}
+		r.healthxHandler = healthx.NewHandler(r.Writer(), r.BuildVersion(), handlers)
```

Comment:
I would rather add a new function that returns a `healthx.ReadyCheckers`, so that you can unit test it and remove some logic from the registry.
```suggestion
		handlers, err := r.HealthEventManager().HealthxReadyCheckers()
		if err != nil {
			r.Logger().WithError(err).Fatalf("Unable to register health checker listener.")
		}
		r.healthxHandler = healthx.NewHandler(r.Writer(), r.BuildVersion(), handlers)
```